lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -654,6 +654,24 @@ protected function getPreAlterTableIndexForeignKeySQL(TableDiff $diff)
             }
         }
 
+        // handle columns that are removed from changed indexes
+        foreach ($diff->changedIndexes as $chgKey => $chgIndex) {
+            // Dropping primary keys requires to unset autoincrement attribute on the particular column first
+            if ($chgIndex->isPrimary() && $diff->fromTable instanceof Table) {
+
+                foreach ($diff->fromTable->getIndex($chgIndex->getName())->getColumns() as $columnName) {
+                    $column = $diff->fromTable->getColumn($columnName);
+
+                    if ($column->getAutoincrement() === true && in_array($columnName, $chgIndex->getColumns()) === false) {
+                        $column->setAutoincrement(false);
```

Comment:
I'm not sure if it's a good idea mutating the "from" table here as this piece of information might be needed later on by other parts of the table alteration (even if it was not the case at the moment). I'd better go a safe way here. So my suggestion is to either set it to `true` again after the SQL generation part or clone the column.
@Ocramius thoughts?